stl/inc/ranges

```diff
@@ -1908,11 +1908,11 @@ namespace ranges {
 
         template <forward_range _Base>
         struct _Category_base<_Base> {
-            using iterator_category =
-                conditional_t<is_lvalue_reference_v<invoke_result_t<_Fn&, range_reference_t<_Base>>>,
-                    conditional_t<derived_from<_Iter_cat_t<iterator_t<_Base>>, contiguous_iterator_tag>,
-                        random_access_iterator_tag, _Iter_cat_t<iterator_t<_Base>>>,
-                    input_iterator_tag>;
+            using iterator_category = conditional_t<
+                is_lvalue_reference_v<invoke_result_t<_Maybe_const<is_const_v<_Base>, _Fn>&, range_reference_t<_Base>>>,

```

Comment:
I'm not sure whether `is_const_v<_Base>` is OK. It seems possible to have a `const` type satisfying `view`, although I think this is highly impractical.

Reply:
Does `_Vw`'s constness matter? I believe that `invoke_result_t` only cares the constness of `*_Parent->_Fun`, which matches the constmess of `_Parent_t`.